src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs

```diff
@@ -470,13 +482,18 @@ private void InitializeTypeInformation()
 
         private string DebuggerToString()
         {
-            if (Identity.MetadataKind == ModelMetadataKind.Type)
-            {
-                return $"ModelMetadata (Type: '{ModelType.Name}')";
-            }
-            else
+            switch (MetadataKind)
             {
-                return $"ModelMetadata (Property: '{ContainerType.Name}.{PropertyName}' Type: '{ModelType.Name}')";
+                case ModelMetadataKind.Parameter:
+                    return $"ModelMetadata (Parameter: '{ParameterName}' Type: '{ModelType.Name}')";
+                case ModelMetadataKind.Property:
+                    return $"ModelMetadata (Property: '{ContainerType.Name}.{PropertyName}' Type: '{ModelType.Name}')";
+                case ModelMetadataKind.Type:
+                    return $"ModelMetadata (Type: '{ModelType.Name}')";
+                default:
+                    var message = $"Unsupported MetadataKind '{MetadataKind}'.";
+                    Debug.Fail(message);
+                    return message;
```

Comment:
Since this will be evaluated by the debugger, I don't think Debug.Fail will do good things here.

Reply:
What are you suggesting instead? Skip the `default` case,`throw` an `InvalidOperationException`, or something else?

Reply:
Just don't throw or assert. You'll be broken in all kinds of ways in product code if something happens. This will just make it harder to debug.

Reply:
FYI I updated this and `ModelBinderFactory.Key.ToString()` to just return the `message` in their `default:`s.